docs: rewrite Quick Start as short developer guide (CTORNDGAIN-1301) (#82)
griddynamics/rosetta: FAQ.md

# FAQ

**Who is this for?** Anyone evaluating, installing, or using Rosetta who has a quick question that isn't a bug or setup issue.
**When should I read this?** Before opening an issue — most meta-questions are answered here. For setup problems see [INSTALLATION.md](INSTALLATION.md); for things that aren't working see [TROUBLESHOOTING.md](TROUBLESHOOTING.md).

---

## Installation & Detection

**How do I know if Rosetta is installed and configured for this repo?**

Four signals, in order of reliability:

> 1. After you send your first prompt in a new chat session, the agent's reply opens with `I have loaded context using Rosetta: ...`. This is the definitive sign — Rosetta successfully bootstrapped for the session. (Subsequent prompts in the same session reuse the loaded context and won't repeat the line.)

Reviewer: It is different:
1. AI coding agent loads CONTEXT.md and ARCHITECTURE.md files - if not it misses the entire context
2. AI coding agent loads workflow - if not it is taking shortcut delivering subpar results
3. AI coding agent loads hitl and orchestrator-contract - if not it will not engage user and hallucinate

3. A bootstrap rule file exists in the repo (e.g. `CLAUDE.md`, `AGENTS.md`, `.cursor/rules/`, or your IDE's instruction file) and contains the Rosetta bootstrap block. This is the fallback when MCP isn't available.
4. The Rosetta entry exists in your IDE's configuration. The [Uninstalling section in INSTALLATION.md](INSTALLATION.md#uninstalling) lists exactly where Rosetta lives for each IDE and mode — invert it to check for presence:
   - **Plugin:** run `claude plugin list` (Claude Code), check installed Copilot agents (VS Code), or look for extracted plugin files in the repo (Codex)
   - **MCP:** run `claude mcp list` / `codex mcp list`, or open your IDE's MCP config file (Cursor, VS Code, Windsurf, JetBrains, Antigravity, OpenCode) and look for a `Rosetta` entry
   - **Offline:** confirm an `instructions/` directory exists in the repo

If none of these are true, Rosetta is not active for this session. See [INSTALLATION.md](INSTALLATION.md).

**How do I install Rosetta for the first time?**

> See the [Get Started section in README.md](README.md#get-started) for the fastest path, or [INSTALLATION.md](INSTALLATION.md) for the full setup including the fallback bootstrap rule. Once installed, [QUICKSTART.md](QUICKSTART.md) walks you through your first session.

Reviewer: ```suggestion
See the [Quick Start Guide](QUICKSTART.md) for the fastest path, or [INSTALLATION.md](INSTALLATION.md) for the full setup including the fallback bootstrap rule. Once installed, [QUICKSTART.md](QUICKSTART.md) walks you through your first session.
```

**Which Rosetta release should I use?**

**R2** is the current stable release — use it for production work.

> **R3** is in active development and not for production use yet. (Internal contributors testing R3 do so by switching to the `V3` branch — see [DEVELOPER_GUIDE.md](DEVELOPER_GUIDE.md).)

Reviewer: R3/V3 branch and so on - remove, this is not for users of rosetta

> Rosetta uses release-based versioning so new instructions can be developed without breaking agents on stable releases, and rollback is always possible. See [OVERVIEW.md](OVERVIEW.md) for the rationale.

Reviewer: Rosetta supports N-1 releases

**How do I upgrade from R1 to R2?**

Open a new chat in your IDE and type: `Initialize this repository using Rosetta (upgrade R1 to R2)`. Rosetta will detect the existing R1 layout and migrate it.

**Plugin install or MCP install — which should I use?**

Use the **plugin** when one is available for your IDE. Plugins bundle the bootstrap rule, skills, agents, and workflows directly into the IDE and the agent loads them locally — no live connection to a Rosetta server at request time. In practice this means:

- Faster session start, no MCP OAuth expirations dropping the agent mid-task, no dependency on the Rosetta server being reachable from your network.

> - No outbound calls to an external service means no data-egress or privacy review — you're installing static instruction files into the repo. That's a much shorter conversation with security and compliance than authorizing a new external MCP endpoint.

Reviewer: ```suggestion
- No outbound calls to an external service means no data-egress or privacy review — you're using local static instruction files. That's a much shorter conversation with security and compliance than authorizing a new external MCP endpoint.
```

Use the **MCP** install when no plugin path exists for your IDE — e.g. Windsurf, Antigravity, OpenCode, or JetBrains Junie. See [PLUGINS.md](PLUGINS.md) for the IDEs that currently ship a plugin, and [INSTALLATION.md](INSTALLATION.md) for MCP setup.

---

## Token Usage & Performance

**Does an AI agent consume more tokens with Rosetta?**

Yes, but with a purpose. Rosetta loads bootstrap rules, the workflow for your request type, and any skills needed for the task — typically a few thousand extra tokens up front. In return you get:

> - Fewer wrong-path executions that would burn far more tokens being undone

Reviewer: Guardrails, Security, Sensitive Data handling, current config risk assessment, reliable results.

- Less back-and-forth because the agent asks the right questions early (HITL)

> - Reusable plans, specs, and memory that compound across sessions

Reviewer: ```suggestion
- Spec-Driven Development extended with Discovery, Design, Review, and Validation phases
- Reusable agent memory that allows AI to learn on current repo
```

> For small/trivial tasks the overhead can feel high relative to the task. For medium/large tasks the overhead is small relative to total work and the reliability gain usually pays for itself. <!-- TODO: link to any published benchmarks if available -->

Reviewer: ```suggestion
For small/trivial tasks AI treats them as just small change and never executes tests nor validate results of changes. Very often small changes have side effects, which AI must figure out first. This leads to low quality unpredictable results.
```
